schema/proto3/port.proto

```diff
@@ -28,13 +28,14 @@ message PortConfiguration {
     
```

Comment:
@xieus - I didn't add the "message version/timestamp" at this point to handle out of order message. With multiple DPMs running for scale, out of order message could happen. Should we go ahead and add "uint32 message_version" for our "GoalState" message?

Reply:
I would say that is the way to go. My recommendation is:
* use the existing version, or rename it to "content_version" to track the content version of the message
* use "format_version" or some better name to track the formatting version of the message (e.g. adding a new field)

Reply:
would the content_version be on the PortConfiguration level or on the GoalState level? I can see how we would do it at the GoalState level. Putting it on GoalState level would be complicated because we can have multiple PortConfiguration in one GoalState message.

Reply:
I think port level is appropriate.

Reply:
ok, thinking about it on the route, I think we may need to use port ID and content_version as the "key" to determine the actual content_version since different port IDs on PortConfigurations won't conflict with each other right?

how do we deal with the below GoalState message which have?
- PortConfiguration - portID = 1, content_version = 5
- PortConfiguration - portID = 1, content_version = 7
- PortConfiguration - portID = 1, content_version = 6

do we process the PortConfiguration in the order of 5, 6, 7? Note that we have parallel processing in ACA, so maybe we cannot go parallel when we detect there are multiple PortConfigurations with the same port ID within the same GoalState message?

Reply:
This depends on whether the port configuration is full or delta. When comes to delta, out-of-order processing may or may not have impact to the results depending on scenarios. So it is important to identify those scenarios. 

Regarding parallel processing, the answer is related to scenarios as well.

Reply:
ok, I will think about the scenarios a bit more.